test/storage/file.js

```diff
@@ -1025,6 +1025,22 @@ describe('File', function() {
     });
   });
 
+  describe('makePublic', function() {
+    it('should make the file public', function(done) {
+
+      // If this mocked method (THAT DOES NOTHING!) is removed, nothing works.
+      file.acl.makeAclObject_ = function() {};
```

Comment:
Quick answer: You should be testing that when you call `file.makePublic` it calls `this.acl.add` with the correct arguments.

Philosophical explanation: It's not the business of this test to be concerned with how `this.acl.add` does its job - we have tests for that. That's why you have to mock `file.acl.makeAclObject_`, because `this.acl.add` is using that method to do its job. And again, our test of `makePublic` only cares that `this.acl.add` is getting what it needs, and if we trust our acl test suite, then we don't need to question it from way up here.